plugins/extrusions/extrusions.py

```diff
@@ -0,0 +1,147 @@
+    """
+    Return a 2080 extrusion with the specified length
+
+    :param length: length of the extrusion
+    :type length: float > 0
+    :param centered: If True, the extrusion will be centered around the reference point.
+        If False, the corner of the extrusion will be on the reference point and it will
+        extend in the positive x, y and z directions. Can also use a 3-tuple to
+        specify centering along each axis.
+    """
+    return _extrusion([80, 20], "e2080.dxf", length, centered)
+
+
+def e4040(length: float, centered: Union[bool, Tuple[bool, bool, bool]] = True) -> T:
+    """
+    Return a 4040 extrusion with the specified length
+
+    :param length: length of the extrusion
+    :type length: float > 0
+    :param centered: If True, the extrusion will be centered around the reference point.
+        If False, the corner of the extrusion will be on the reference point and it will
+        extend in the positive x, y and z directions. Can also use a 3-tuple to
+        specify centering along each axis.
+    """
+    return _extrusion([40, 40], "e4040.dxf", length, centered)
+
+
+def v2020(length: float, centered: Union[bool, Tuple[bool, bool, bool]] = True) -> T:
+    """
+    Return a 2020 v-slot extrusion with the specified length
+
+    :param length: length of the extrusion
+    :type length: float > 0
+    :param centered: If True, the extrusion will be centered around the reference point.
+        If False, the corner of the extrusion will be on the reference point and it will
+        extend in the positive x, y and z directions. Can also use a 3-tuple to
+        specify centering along each axis.
+    """
+    return _extrusion([20, 20], "v2020.dxf", length, centered)
+
+
+def v2040(length: float, centered: Union[bool, Tuple[bool, bool, bool]] = True) -> T:
+    """
+    Return a 2040 v-slot extrusion with the specified length
+
+    :param length: length of the extrusion
+    :type length: float > 0
+    :param centered: If True, the extrusion will be centered around the reference point.
+        If False, the corner of the extrusion will be on the reference point and it will
+        extend in the positive x, y and z directions. Can also use a 3-tuple to
+        specify centering along each axis.
+    """
+    return _extrusion([40, 20], "v2040.dxf", length, centered)
+
+
+def v4040(length: float, centered: Union[bool, Tuple[bool, bool, bool]] = True) -> T:
+    """
+    Return a 4040 v-slot extrusion with the specified length
+
+    :param length: length of the extrusion
+    :type length: float > 0
+    :param centered: If True, the extrusion will be centered around the reference point.
+        If False, the corner of the extrusion will be on the reference point and it will
+        extend in the positive x, y and z directions. Can also use a 3-tuple to
+        specify centering along each axis.
+    """
+    return _extrusion([40, 40], "v4040.dxf", length, centered)
```

Comment:
I had an idea for how to reduce the repetition here. From a user perspective this doesn't change anything, it's just a refactoring. 

If you find this too complicated just to reduce some minor repetition that's completely justified and you are welcome to ignore this suggestion.

```suggestion
def _extrusion(
    size: Tuple[float, float],
    filename: str,
    doc_name: str,
) -> Callable:
    """
    Return a function that will in turn generate the standard extrusion.

    :param size: size of the extrusion, used for centering
    :param filename: filename of the extrusion's profile
    :param doc_name: a string that will be inserted into a docstring template and assigned
      to __doc__ on the returned function
    """
    file = pkg_resources.resource_filename(__name__, "profiles/" + filename)
    profile = importers.importDXF(file).wires()
    docstring = f"""
    Return a {doc_name} extrusion with the specified length

    :param length: length of the extrusion
    :type length: float > 0
    :param centered: If True, the extrusion will be centered around the reference
      point.  If False, the corner of the extrusion will be on the reference point and
      it will extend in the positive x, y and z directions. Can also use a 3-tuple to
      specify centering along each axis.
    """

    def rv(length: float, centered: Union[bool, Tuple[bool, bool, bool]]) -> cq.Workplane:

        if isinstance(centered, bool):
            centered = (centered, centered, centered)

        offset = Vector(
            0 if centered[0] else size[0] / 2,
            0 if centered[1] else size[0] / 2,
            -length / 2 if centered[2] else 0,
        )

        extrusion = profile.toPending()._extrude(length).move(Location(offset))
        return cq.Workplane("XY").newObject([extrusion])

    rv.__doc__ = docstring
    return rv


e2020 = _extrusion([20, 20], "e2020.dxf", "2020")
e2040 = _extrusion([40, 20], "e2040.dxf", "2040")
e2080 = _extrusion([80, 20], "e2080.dxf", "2080")
e4040 = _extrusion([40, 40], "e4040.dxf", "4040")
v2020 = _extrusion([20, 20], "v2020.dxf", "2020 v-slot")
v2040 = _extrusion([40, 20], "v2040.dxf", "2040 v-slot")
v4040 = _extrusion([40, 40], "v4040.dxf", "4040 v-slot")
```

Note that you can't use evaluated strings in a function docstring, ie. 
```python
s = "a string"

def foo():
    f"""
    a function relating to {foo}.
    """
    pass
```

Will actually result in `foo.__doc__` being `None`. So instead I modify the docstring after creating the function.

Reply:
OK, I'll look into making that change.